packages/langium/test/lsp/completion/completion-provider.test.ts

```diff
@@ -0,0 +1,24 @@
+import { ParserRule, LangiumDocument, createLangiumGrammarServices, LangiumDocumentConfiguration } from '../../../lib';
+
+describe('Completion Provider', () => {
+    test('case insensitive prefix matching', () => {
+        const services = createLangiumGrammarServices();
+        const model = `grammar g
+        Aaaa: name="A";
+        aaaa: name="a";
+        Bbbb: name="B";
+        C: a=aa;`
+        const document = LangiumDocumentConfiguration.create('', 'langium', 0, model);
+        const parser = services.parser.LangiumParser;
+        const parseResult = parser.parse(model);
+        const rootNode = parseResult.value;
+        (rootNode as { $document: LangiumDocument }).$document = document;
+        document.parseResult = parseResult;
+        document.precomputedScopes = services.references.ScopeComputation.computeScope(document);
```

Comment:
You can simplify the `document` handling, by passing `document` directly into the `parse` method. It will then assign the `$document` in there.